utils.go

```diff
@@ -120,12 +124,25 @@ func makeDataTransfer(ctx context.Context, host host.Host, ds datastore.Batching
 		log.Errorf("Failed to register legs validator voucher type: %s", err)
 		return nil, nil, "", err
 	}
-	if err := dt.RegisterVoucherResultType(lvr); err != nil {
+	if err = dt.RegisterVoucherResultType(lvr); err != nil {
 		log.Errorf("Failed to register legs voucher result: %s", err)
 		return nil, nil, "", err
 	}
-	if err := dt.Start(ctx); err != nil {
-		log.Errorf("Failed to start datatransfer: %s", err)
+
+	if tmpDir != "" {
+		if err = dt.Start(ctx); err != nil {
+			log.Errorf("Failed to start datatransfer: %s", err)
+			return nil, nil, "", err
+		}
+	}
+
+	// Wait for datatrnasfer to be ready.
+	dtReady := make(chan error)
+	dt.OnReady(func(e error) {
+		dtReady <- e
```

Comment:
Does this work if you pass in a dt.Manager that is already started? I think not.

Reply:
Thank you. Fixed